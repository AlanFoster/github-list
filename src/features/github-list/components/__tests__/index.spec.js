import React from 'react';
import List from '../';
import renderer from 'react-test-renderer';

describe('github-list', function () {
  describe('when there is no data available', function () {
    it('renders no rows or indicator', function () {
      const repositories = [];

      const tree = renderer.create(
        <List repositories={repositories} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('when there 2 rows of data available', function () {
    it('renders both rows', function () {
      const repositories = [
        {
          id: 1,
          name: 'first-name',
          fullName: 'first-full-name',
          description: 'first-full-description',
          htmlUrl: 'first-html-url'
        },
        {
          id: 2,
          name: 'second-name',
          fullName: 'second-full-name',
          description: 'second-full-description',
          htmlUrl: 'second-html-url'
        }
      ];

      const tree = renderer.create(
        <List repositories={repositories} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  xdescribe('when there are multiple rows...', function () {

  });
});
