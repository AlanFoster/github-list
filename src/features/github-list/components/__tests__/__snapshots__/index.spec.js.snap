// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`github-list when there 2 rows of data available renders both rows 1`] = `
<div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <div>
        first-full-name
      </div>
      <div>
        first-full-description
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <div>
        second-full-name
      </div>
      <div>
        second-full-description
      </div>
    </div>
  </div>
</div>
`;

exports[`github-list when there is no data available renders no rows or indicator 1`] = `<div />`;
